daisy_workflows/build-publish/rhui/health_check.py

```diff
@@ -0,0 +1,131 @@
+#  Copyright 2022 Google Inc. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Health check for RHUI infrastructure nodes."""
+
+import argparse
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+import typing
+
+
+class HealthChecks:
+  nfs_mount: str
+
+  def __init__(
+      self,
+      nfs_mount: str,
+  ) -> None:
+    self.nfs_mount = nfs_mount
+
+  def mount_uses_nfs(self, logger: logging.Logger):
+    logger.info("checking that %s is mounted as NFS", self.nfs_mount)
+    # Using pipes to capture output since RHEL8 has Python 3.6
+    # which doesn't support `capture_output`.
+    p = subprocess.run(
+      ["stat", "-fc", "%T", self.nfs_mount],
+      stdout=subprocess.PIPE,
+      stderr=subprocess.PIPE,
+    )
+    if p.returncode != 0:
+      raise EnvironmentError("Failed to stat %s: %s" % (self.nfs_mount, p))
```

Comment:
maybe a nit since i'm suggesting not to shell out but subprocess.run can automatically raise on nonzero

Reply:
Not used anymore after change to `/proc/mounts`